Wikipedia/Code/WMFFeedContentSource.m

```diff
@@ -229,15 +229,49 @@ - (void)saveGroupForPictureOfTheDay:(WMFFeedImage *)image date:(NSDate *)date in
     if (image == nil || date == nil) {
         return;
     }
+    
+    NSString *primaryLanguage = self.userDataStore.languageLinkController.appLanguage.contentLanguageCode;
+    NSString *feedLanguage = [self.siteURL wmf_contentLanguageCode];
+    BOOL isPrimaryLanguageFeed = [feedLanguage wmf_isEqualToStringIgnoringCase:primaryLanguage];
+    
+    // T356255 - the POTD is the same across all feeds (i.e. same image, different localized description).
+    // To prevent the same image from showing on each language feed, we save the POTD for the primary language only
+    // (and clean up any secondary / duplicate ones - secondary ones may exist when the user has removed a language
+    // or changed the app primary language).
+    
+    if (isPrimaryLanguageFeed) {
+        [self insertOrUpdatePictureOfTheDayForImage:image date:date inManagedObjectContext:moc];
+        [self removeDuplicatePicturesOfTheDayForDate:date inManagedObjectContext:moc];
+    }
+}
```

Comment:
**Change 2 of 3**

This now inserts (or updates) POTD against the primary language only (and removes any duplicate ones found against secondary languages).